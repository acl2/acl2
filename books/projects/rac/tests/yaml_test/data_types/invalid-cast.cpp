#include <array>
using namespace std;

// RAC begin

int foo(array<int, 3> a) { return (int)a; }

// RAC end
