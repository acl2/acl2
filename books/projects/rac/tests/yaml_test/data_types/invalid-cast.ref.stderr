invalid-cast.cpp:6 (34-40): Incompatible types: int[3] cannot be cast to int
  6 | int foo(array<int, 3> a) { return (int)a; }
    |                                   ^^^^^^
note: To convert types depending on template parameters to primitive types, you should add an explicit conversion (like this ac_int<32, true>(...) to convert to int)
